Report/Template.Rmd

```diff
@@ -243,17 +251,267 @@ datasets = read.and.clean(dataset = "easySHARE_rel6_0_0.rda", wav = 99)
+agegroups = names(df.out[, 4:6])
+args      = with(df.out, 
+                 expand.grid(levels(country), levels(gender), agegroups, 
+                             stringsAsFactors = FALSE))
+args.list = list(c = as.vector(t(args[1])), 
+                 g = as.vector(t(args[2])), 
+                 a = as.vector(t(args[3])))
+
+# Map over multiple arguments
+output   = pmap(args.list, function(c, g, a) labor(df.out, c, g, a))
+output.v = do.call("c", output)
+
+labor.part.share    = cbind(args, output.v)
+labor.part.share.df = spread(labor.part.share, Var3, output.v) %>% 
+    arrange(desc(Var2)) %>% 
+    set_colnames(c("Country", "Gender", 
+                   paste0("Age ", rep(c("50-54", "55-59", "60-64")))))
+```
+
+The results can be easily bound into a data frame and arranged using `dplyr`. The same package is used to subset by gender. Using the pipe operator `%>%` keeps our code compact and avoids unnecessary variable assignment and inefficient memory requirements.
+
+`vars` is actually a name vector that specifies whether labor force participation is full time, part time or not prevalent. Fetching it using `grep` from the cleaned data set makes it robust against name changes and ensures consistency. Conveniently, `grep` does not necessarily require regular expressions as an input. Instead, it is able to perform matches on simple strings and, in this case, returns the names that contain 'labor'. 
+
+```{r, eval = FALSE}
+
+# Get gender specific tables
+labor.part.f = labor.part.share.df %>% 
+    filter(Gender == "FEMALE") %>% 
+    set_rownames(levels(df.out$country)) %>% 
+    dplyr::select(-Gender, -Country)
+
+# Labor supply choice tables
+df.female = df.out %>% filter(gender == "FEMALE")
+vars      = names(df.out)[grep("labor", names(df.out))][3:1] # invert
+
+labor.supply.f    = lapply(vars, function(x) group.share(df.female, x, 1))
+labor.supply.f.df = do.call(cbind.data.frame, labor.supply.f) %>% 
+    set_colnames(c("Nonparticipation", "Half time", "Full time"))
+
+
+```
+
+
+
+
+## Empirical Results
+
+```{r, eval = TRUE, results = 'asis', warning = FALSE, comment = FALSE, message = FALSE, echo = FALSE}
+
+library("knitr")
+knitr::kable(t(round(sum.stats.tot, 2)), caption = "Summary Statistics over all groups")
+
+
+knitr::kable(labor.part.m, caption = "Men: Labor Force Participation Rate by age")
+knitr::kable(labor.part.f, caption = "Women: Labor Force Participation Rate by age")
+
+```
+
+
+Having obtained the share of labor force participation grouped by gender, country and age as well as the according health variables, striking values can be presented.
+
+Among the selected countries, France, Greece and Belgium have the fewest people among the oldest age group, which is ranging from 60 to 64 years. In the middle age group, ranging from 54-59, the German population is sparsely represented. Within the youngest age group, including people from 50 to 54 years old, Italy and Austria have the lowest share.  
```

Comment:
Id be careful here - this are not really any results but mere sample artifacts.

Reply:
Good catch